kratos/includes/registry_auxiliaries.h

```diff
@@ -0,0 +1,195 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//                   Pooyan Dadvand
+//
+
+#pragma once
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+
+// Project includes
+#include "includes/registry.h"
+
+namespace Kratos
+{
+
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @brief Kratos registry auxiliaries
+ * This static class collect all the auxiliary functions to be used to register c++ items
+ */
+class RegistryAuxiliaries final
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of Registry
+    KRATOS_CLASS_POINTER_DEFINITION(RegistryAuxiliaries);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    RegistryAuxiliaries() = delete;
+
+    /// Copy constructor.
+    RegistryAuxiliaries(RegistryAuxiliaries const& rOther) = delete;
+
+    /// Destructor.
+    ~RegistryAuxiliaries() = delete;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    template<typename TPrototypeType>
+    static void RegisterProcessWithPrototype(
```

Comment:
Are we gonna allow `Process` also registered via Prototypes and Create methods?

Reply:
We don't, but we name it just in case.